tests/extra_python_package/test_files.py

```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+import subprocess
+import sys
+import os
+import tarfile
+import contextlib
+import zipfile
+
+# These tests must be run explicitly
+# They require CMake 3.15+ (--install)
```

Comment:
`--install` can also be achieved by `--target install`, no? But I suppose your point is that this is internal to development & CI anyway, so we can require a recent CMake version?

Reply:
CMake is _only_ required if you are building from the GitHub sources, it is not required from the `--sdist`. I like `--install` better than `--target install`, because it does not run the underlying build system; but to your point, since we need an underlying build system here for CMake to configure, we could reduce the requirements if there's a desire to.

Reply:
The real requirement here is CMake 3.14+ since I use the `-S <> -B <>` syntax, as it's simpler and OS agnostic. Making that 3.15+ didn't seem like a big jump. I could add the manual path handling and make this 3.4+ though.

Reply:
Nah, don't worry! I think it's good enough, like this. Just wanted to make sure. But it is only for the CI/developers, right?

Reply:
Yes. Only used if you are _building the python package_ from GitHub, and not PyPI. PyPI, even the sdist, has the rendered CMake target/helper files.